### PR TITLE
jobs/build: cut image building parallelism in half

### PR DESCRIPTION
**Pull request by jlebon**
We're seeing issues where the pipeline sometimes hits EAGAIN on `fork`:

https://github.com/coreos/fedora-coreos-tracker/issues/1186

This may have started happening more frequently after the recent
addition of VirtualBox.

Anyway, it's kinda silly to build that many images in parallel. There's
diminishing returns and eventually a performance cost. So let's do it in
two rounds instead. (Ideally instead it'd be a kind of task pool, but
let's not try to implement that in Groovy.)

### CODE REVIEW

**Review by cgwalters — APPROVED** [review]
Looks sane, but I think what we really want is to have `cosa buildextend -j N` or so...

**Comment by dustymabe** [comment]
Hmm. Let's be a bit smarter here and put the ones that are long running together. Let me look at the run history and make a suggestion.

**Comment by jlebon** [comment]
> Hmm. Let's be a bit smarter here and put the ones that are long running together. Let me look at the run history and make a suggestion.

I'm not sure it's worth trying to optimize this further vs e.g. pooling as suggested. The step itself only takes ~5 minutes which is tiny in comparison to a full pipeline run (~1h30m). It'd make adding new platforms trickier too.

**Comment by dustymabe** [comment]
> I'm not sure it's worth trying to optimize this further vs e.g. pooling as suggested. The step itself only takes ~5 minutes which is tiny in comparison to a full pipeline run (~1h30m). It'd make adding new platforms trickier too.

Yeah OK. I vaguely remembered the DigitalOcean one taking more than 10 minutes at one point in the past and one other one being long too and was going to try to put them in the same parallel bucket. Looking at the most recent runs it seems like the times are really too bad.

File: jobs/build.Jenkinsfile [hunk 1]
```diff
@@ -418,17 +418,19 @@ lock(resource: "build-${params.STREAM}") {
                 }
             }
 
-            // parallel build these artifacts
-            def pbuilds = [:]
-            ["Aliyun", "AWS", "Azure", "AzureStack", "DigitalOcean", "Exoscale", "GCP", "IBMCloud", "Nutanix", "OpenStack", "VirtualBox", "VMware", "Vultr"].each {
-                pbuilds[it] = {
+            // parallel build these artifacts over two runs
```
Comment (dustymabe) [inline jobs/build.Jenkinsfile:421]: The first question I have in the future when I read this code is "why?". Why split it over two runs when no steps depend on each other?

We know the answer now.. Let's add a comment explaining?

**Review by dustymabe — APPROVED** [review]
LGTM

**Comment by dustymabe** [comment]
Thanks for working on this.

**Pull request merged**